package.json

```diff
@@ -205,7 +205,7 @@
     "ts-dedent": "^2.0.0",
     "ts-jest": "^26.4.4",
     "ts-node": "^8.10.2",
-    "typescript": "^3.9.7",
+    "typescript": "^4.1.3",
```

Comment:
@ndelangen @gaetanmaisse are we ok upgrading the typescript version?

Reply:
for ivy and ngcc typescript 4 is prerequisite 
https://github.com/angular/angular/blob/master/packages/compiler-cli/package.json#L32

I think it's added here just for angular test with ivy
but that this should be the subject of a dedicated PR. it seems to me that this breaks some build

Reply:
> But that this should be the subject of a dedicated PR. it seems to me that this breaks some build

Yep definitely, I'm not sure it will break the build but there are some references to update (look for `ts3.9` in the monorepo). Let's discuss that with @ndelangen on Discord 😉

Reply:
> I think it's added here just for angular test with ivy
☝️ 

But it is more an internal blocker. Angular apps will have the correct combination of typescript and angular/compiler so `ngcc` will either have the workaround for ts 3.9 or the project uses typescript >= 4.0 anyway 

Internally, we need to have typescript >= 4.0 because the latest Angular 11 version is installed. `ngcc` won't run in  typescript 3.9

Reply:
I remember `downlevel-dts` didn't support typescript 4 yet, last time I tried updating.. If that's been solved, we can proceed IMHO.

Us depending on typescript 4 isn't a breaking change for users, I assume?

Reply:
Shouldn't be! 

For reference: https://github.com/sandersn/downlevel-dts/pull/51

Reply:
Cool then we can proceed, with the upgrade I think!